src/Integration.Vsix.UnitTests/Commands/ProjectTestPropertySetCommandTests.cs

```diff
@@ -0,0 +1,454 @@
+            command.Enabled = true;
+
+            var project = new ProjectMock("project.csproj");
+            project.SetCSProjectKind();
+            this.projectSystem.SelectedProjects = new[] { project };
+
+            // Test case 1: test (true)
+            // Setup
+            var testSubject1 = new ProjectTestPropertySetCommand(serviceProvider, true);
+
+            // Act
+            testSubject1.Invoke(command, null);
+
+            // Verify
+            this.VerifyTestProperty(project, true);
+
+            // Test case 2: non-test (false)
+            var testSubject2 = new ProjectTestPropertySetCommand(serviceProvider, false);
+
+            // Act
+            testSubject2.Invoke(command, null);
+
+            // Verify
+            this.VerifyTestProperty(project, false);
+
+            // Test case 3: auto detect (null)
+            var testSubject3 = new ProjectTestPropertySetCommand(serviceProvider, null);
+
+            // Act
+            testSubject3.Invoke(command, null);
+
+            // Verify
+            this.VerifyTestProperty(project, null);
+        }
+
+        [TestMethod]
+        public void ProjectTestPropertySetCommand_Invoke_MultipleProjects_MixedPropValues_SetsValues()
+        {
+            // Setup
+            OleMenuCommand command = CommandHelper.CreateRandomOleMenuCommand();
+            command.Enabled = true;
+
+            var p1 = new ProjectMock("good1.proj");
+            var p2 = new ProjectMock("good2.proj");
+            p1.SetCSProjectKind();
+            p2.SetCSProjectKind();
+            this.projectSystem.SelectedProjects = new[] { p1, p2 };
+
+            // Test case 1: test (true)
+            // Setup
+            var testSubject1 = new ProjectTestPropertySetCommand(serviceProvider, true);
+            this.SetTestProperty(p1, null);
+            this.SetTestProperty(p2, true);
+
+            // Act
+            testSubject1.Invoke(command, null);
+
+            // Verify
+            this.VerifyTestProperty(p1, true);
+            this.VerifyTestProperty(p2, true);
+
+            // Test case 2: non-test (false)
+            var testSubject2 = new ProjectTestPropertySetCommand(serviceProvider, false);
+            this.SetTestProperty(p1, true);
+            this.SetTestProperty(p2, false);
+
+            // Act
+            testSubject2.Invoke(command, null);
+
+            // Verify
+            this.VerifyTestProperty(p1, false);
+            this.VerifyTestProperty(p2, false);
+
+            // Test case 3: auto detect (null)
+            var testSubject3 = new ProjectTestPropertySetCommand(serviceProvider, null);
+            this.SetTestProperty(p1, true);
+            this.SetTestProperty(p2, null);
+
+            // Act
+            testSubject3.Invoke(command, null);
+
+            // Verify
+            this.VerifyTestProperty(p1, null);
+            this.VerifyTestProperty(p2, null);
+        }
+
+        [TestMethod]
```

Comment:
`InvokeInternal` can't be called unless `QueryStatusInternal` set the command to be enabled.
See [here](https://github.com/SonarSource-VisualStudio/sonarlint-visualstudio/blob/master/src/Integration.Vsix/Commands/VsCommandBase.cs#L41).
There are also debug assert messages in the `InvokeInternal` methods (for both commands), and even if `InvokeInternal` is executed without any selected projects, nothing happens. It's pretty hard to test if nothing external changes with no input.